src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -266,6 +301,9 @@ object Directory {
       }
     }
 
+    if(fileNameToIdMap.isDefined) {
+      fileNameToIdMap.get.put(IndexConstants.MAX_FILE_ID, maxId)
+    }
```

Comment:
Instead of this, does it make sense to have a proper class to capture this max id and other adding logic? For example,
```scala

class FileIdTracker(private val startId: Long) {
  private var maxId: Long = startId

  private var fileNameToIdMap: mutable.HashMap[String, Long] = mutable.HashMap()

  def getFileNameToIdMap: HashMap[String, Long] = ??? // Use fileNameToIdMap

  def getMaxFileId: Long = maxId

  // Expose different APIs to add files. For example:

  def addFileStatusSeq(files: Seq[FileStatus]): Unit = ???
  def addFileInfoSeq(files: Seq[FileInfo]): Unit = ???
}
```

Reply:
I added it under `IndexLogEntry` . Please take a look.